### ISSUE TITLE
everything's screwed up

### ISSUE THREAD

**Issue by bp88dev** [Rapptz/discord.py#9992]
### Summary

Things aren't working how they should be

### Reproduction Steps

I simply left my code alone for a bit, running when suddenly it started erroring out. I never changed any of the code I'm using. I then tried to remediate the problem but then discovered that my interactions were timing out. No big deal, but then I tried using both edit commands, they didn't work. I eventually got it to work, but FFMPEG OF ALL THINGS CRASHES. I have verified that my FFmpeg exe is working, because I haven't changed it or anything since it was working.

### Minimal Reproducible Code

```python
@tree.command(name="test", description="Run whatever code is on the chopping board.")
async def test(interaction):
    try:
        interaction.user.voice.channel
    except:
        await interaction.response.send_message(embed=discord.Embed(title="You are not in a voice channel!", color=discord.Color.red()))
        return
    for i in interaction.guild.channels:
        for i2 in connections:
            if i == i2["channel"]:
                embed = discord.Embed(title="I'm already in a channel!", color=discord.Color.red())
                await interaction.response.send_message(embed=embed)
                return
    #message = await interaction.response.send_message(embed=discord.Embed(title="Connecting to voice channel...", color=discord.Color.light_gray()))
    await interaction.response.send_message(embed=discord.Embed(title="Connected to voice channel.", color=discord.Color.green()))
    voice = await interaction.user.voice.channel.connect()
    connections.append({"voiceObject": voice, "channel": interaction.user.voice.channel})
    #await interaction.response.send_message(embed=discord.Embed(title="Connected to voice channel.", color=discord.Color.green()))
    #await interaction.message.edit(embed=discord.Embed(title="Connected to voice channel.", color=discord.Color.green()))
    try:
        while True:
            voice.play(discord.FFmpegPCMAudio(executable="ffmpeg.exe", source="test.mp3"))
            await asyncio.sleep(int(mp3.MP3("test.mp3").info.length))
            while voice.is_playing():
                await asyncio.sleep(0.05)
    except:
        if voice.is_connected():
            await retry(voice, "test.mp3")
        else:
            return
```

### Expected Results

The code should connect to a voice channel, and play test.mp3, but it's not doing that.

### Actual Results

```
2024-10-29 20:01:32 ERROR    discord.app_commands.tree Ignoring exception in command 'test'
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\contextlib.py", line 586, in __exit__
    if cb(*exc_details):
       ^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\contextlib.py", line 469, in _exit_wrapper
    callback(*args, **kwds)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 221, in Close
    CloseHandle(self)
RecursionError: maximum recursion depth exceeded while calling a Python object

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\contextlib.py", line 586, in __exit__
    if cb(*exc_details):
       ^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\contextlib.py", line 469, in _exit_wrapper
    callback(*args, **kwds)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 221, in Close
    CloseHandle(self)
RecursionError: maximum recursion depth exceeded while calling a Python object

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\Benji\Documents\portalradio\main.py", line 10, in retry
    voice.play(discord.FFmpegPCMAudio(executable="ffmpeg.exe", source=audio))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 290, in __init__     
    super().__init__(source, executable=executable, args=args, **subprocess_kwargs)
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 166, in __init__     
    self._process = self._spawn_process(args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 180, in _spawn_process
    process = subprocess.Popen(args, creationflags=CREATE_NO_WINDOW, **subprocess_kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 1553, in _execute_child
    self._close_pipe_fds(p2cread, p2cwrite,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 1289, in _close_pipe_fds
    with contextlib.ExitStack() as stack:
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\contextlib.py", line 601, in __exit__
    raise exc_details[1]
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\contextlib.py", line 586, in __exit__
    if cb(*exc_details):
       ^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\contextlib.py", line 469, in _exit_wrapper
    callback(*args, **kwds)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 221, in Close
    CloseHandle(self)
RecursionError: maximum recursion depth exceeded while calling a Python object

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\Benji\Documents\portalradio\main.py", line 10, in retry
    voice.play(discord.FFmpegPCMAudio(executable="ffmpeg.exe", source=audio))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 290, in __init__     
    super().__init__(source, executable=executable, args=args, **subprocess_kwargs)
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 166, in __init__     
    self._process = self._spawn_process(args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 180, in _spawn_process
    process = subprocess.Popen(args, creationflags=CREATE_NO_WINDOW, **subprocess_kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 992, in __init__
    errread, errwrite) = self._get_handles(stdin, stdout, stderr)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 1346, in _get_handles
    with self._on_error_fd_closer() as err_close_fds:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\contextlib.py", line 301, in helper
    return _GeneratorContextManager(func, args, kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\contextlib.py", line 105, in __init__
    self.gen = func(*args, **kwds)
               ^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\Benji\Documents\portalradio\main.py", line 10, in retry
    voice.play(discord.FFmpegPCMAudio(executable="ffmpeg.exe", source=audio))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 290, in __init__     
    super().__init__(source, executable=executable, args=args, **subprocess_kwargs)
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 166, in __init__     
    self._process = self._spawn_process(args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 180, in _spawn_process
    process = subprocess.Popen(args, creationflags=CREATE_NO_WINDOW, **subprocess_kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 992, in __init__
    errread, errwrite) = self._get_handles(stdin, stdout, stderr)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 1346, in _get_handles
    with self._on_error_fd_closer() as err_close_fds:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\contextlib.py", line 301, in helper
    return _GeneratorContextManager(func, args, kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\Benji\Documents\portalradio\main.py", line 10, in retry
    voice.play(discord.FFmpegPCMAudio(executable="ffmpeg.exe", source=audio))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 290, in __init__     
    super().__init__(source, executable=executable, args=args, **subprocess_kwargs)
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 166, in __init__     
    self._process = self._spawn_process(args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 180, in _spawn_process
    process = subprocess.Popen(args, creationflags=CREATE_NO_WINDOW, **subprocess_kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 992, in __init__
    errread, errwrite) = self._get_handles(stdin, stdout, stderr)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 1346, in _get_handles
    with self._on_error_fd_closer() as err_close_fds:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\contextlib.py", line 301, in helper
    return _GeneratorContextManager(func, args, kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded while calling a Python object

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\Benji\Documents\portalradio\main.py", line 10, in retry
    voice.play(discord.FFmpegPCMAudio(executable="ffmpeg.exe", source=audio))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 290, in __init__     
    super().__init__(source, executable=executable, args=args, **subprocess_kwargs)
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 166, in __init__     
    self._process = self._spawn_process(args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 180, in _spawn_process
    process = subprocess.Popen(args, creationflags=CREATE_NO_WINDOW, **subprocess_kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 992, in __init__
    errread, errwrite) = self._get_handles(stdin, stdout, stderr)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 1346, in _get_handles
    with self._on_error_fd_closer() as err_close_fds:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\Benji\Documents\portalradio\main.py", line 10, in retry
    voice.play(discord.FFmpegPCMAudio(executable="ffmpeg.exe", source=audio))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 290, in __init__     
    super().__init__(source, executable=executable, args=args, **subprocess_kwargs)
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 166, in __init__     
    self._process = self._spawn_process(args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 180, in _spawn_process
    process = subprocess.Popen(args, creationflags=CREATE_NO_WINDOW, **subprocess_kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\subprocess.py", line 822, in __init__
    _cleanup()
RecursionError: maximum recursion depth exceeded

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\Benji\Documents\portalradio\main.py", line 10, in retry
    voice.play(discord.FFmpegPCMAudio(executable="ffmpeg.exe", source=audio))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 290, in __init__     
    super().__init__(source, executable=executable, args=args, **subprocess_kwargs)
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 166, in __init__     
    self._process = self._spawn_process(args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 180, in _spawn_process
    process = subprocess.Popen(args, creationflags=CREATE_NO_WINDOW, **subprocess_kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\Benji\Documents\portalradio\main.py", line 10, in retry
    voice.play(discord.FFmpegPCMAudio(executable="ffmpeg.exe", source=audio))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 290, in __init__     
    super().__init__(source, executable=executable, args=args, **subprocess_kwargs)
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 166, in __init__     
    self._process = self._spawn_process(args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 180, in _spawn_process
    process = subprocess.Popen(args, creationflags=CREATE_NO_WINDOW, **subprocess_kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded while calling a Python object

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\Benji\Documents\portalradio\main.py", line 10, in retry
    voice.play(discord.FFmpegPCMAudio(executable="ffmpeg.exe", source=audio))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 290, in __init__     
    super().__init__(source, executable=executable, args=args, **subprocess_kwargs)
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 156, in __init__     
    piping = subprocess_kwargs.get('stdin') == subprocess.PIPE
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded in comparison

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\Benji\Documents\portalradio\main.py", line 10, in retry
    voice.play(discord.FFmpegPCMAudio(executable="ffmpeg.exe", source=audio))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 283, in __init__     
    args.extend(('-f', 's16le', '-ar', '48000', '-ac', '2', '-loglevel', 'warning'))
RecursionError: maximum recursion depth exceeded while calling a Python object

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\Benji\Documents\portalradio\main.py", line 10, in retry
    voice.play(discord.FFmpegPCMAudio(executable="ffmpeg.exe", source=audio))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\app_commands\commands.py", line 828, in _do_call
    return await self._callback(interaction, **params)  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Benji\Documents\portalradio\main.py", line 135, in test
    await retry(voice, "test.mp3")
  File "c:\Users\Benji\Documents\portalradio\main.py", line 16, in retry
    await retry(voice,audio)
  File "c:\Users\Benji\Documents\portalradio\main.py", line 16, in retry
    await retry(voice,audio)
  File "c:\Users\Benji\Documents\portalradio\main.py", line 16, in retry
    await retry(voice,audio)
  [Previous line repeated 979 more times]
  File "c:\Users\Benji\Documents\portalradio\main.py", line 15, in retry
    if voice.is_connected():
       ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\voice_client.py", line 533, in is_connected
    return self._connected.is_set()
           ^^^^^^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\app_commands\tree.py", line 1248, in _call
    await command._invoke_with_namespace(interaction, namespace)
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\app_commands\commands.py", line 853, in _invoke_with_namespace
    return await self._do_call(interaction, transformed_values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\app_commands\commands.py", line 846, in _do_call
    raise CommandInvokeError(self, e) from e
discord.app_commands.errors.CommandInvokeError: Command 'test' raised an exception: RecursionError: maximum recursion depth exceeded
Exception ignored in: <function AudioSource.__del__ at 0x00000216BAC21F80>
Traceback (most recent call last):
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 117, in __del__      
    self.cleanup()
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 226, in cleanup      
    self._kill_process()
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 190, in _kill_process
    proc = self._process
           ^^^^^^^^^^^^^
AttributeError: 'FFmpegPCMAudio' object has no attribute '_process'
Exception ignored in: <function AudioSource.__del__ at 0x00000216BAC21F80>
Traceback (most recent call last):
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 117, in __del__      
    self.cleanup()
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 226, in cleanup      
    self._kill_process()
  File "C:\Users\Benji\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\discord\player.py", line 190, in _kill_process
    proc = self._process
           ^^^^^^^^^^^^^
AttributeError: 'FFmpegPCMAudio' object has no attribute '_process'
```

### Intents

all

### System Information

- Python v3.11.9-final
- discord.py v2.3.2-final
- aiohttp v3.10.5
- system info: Windows 10 10.0.22631

btw, for some reason it says windows 10, but I'm on 11. just goes to show that windows 11 is just a reskin of windows 10 lmao

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

**Comment by AbstractUmbra** [comment]
Hi,

You should update to the latest supported version of `discord.py` (not `discord`) via pip.

```sh
pip uninstall discord
pip install --force-reinstall -U discord.py
```

There have been a few voice related cleanup and fixes since 2.3.2 and this would hopefully resolve the issue presented.
In your case it looks like #9572 is the culprit and this was resolved in 2.4.0 as you can see via the [changelog](https://discordpy.readthedocs.io/en/stable/whats_new.html#bug-fixes).

**Comment by bp88dev** [comment]
> Hi,
> 
> You should update to the latest supported version of `discord.py` (not `discord`) via pip.
> 
> ```shell
> py -m pip uninstall discord
> py -m pip install --force-reinstall -U discord.py
> ```
> 
> NOTE: YMMV on the above commands since you use the Windows Store version of Python.
> 
> There have been a few voice related cleanup and fixes since 2.3.2 and this would hopefully resolve the issue presented. In your case it looks like #9572 is the culprit and this was resolved in 2.4.0 as you can see via the [changelog](https://discordpy.readthedocs.io/en/stable/whats_new.html#bug-fixes).

Thanks so much! The bot is working now!
